paymentsheet/src/test/java/com/stripe/android/paymentsheet/IntentConfirmationHandlerTest.kt

```diff
@@ -1284,22 +1308,282 @@ class IntentConfirmationHandlerTest {
+    fun `On Google Pay succeeded, should proceed to intent confirmation`() = runTest {
+        val intentConfirmationInterceptor = FakeIntentConfirmationInterceptor()
+        val googlePayCallbackHandler = FakeResultHandler<GooglePayPaymentMethodLauncher.Result>()
+        val intentConfirmationHandler = createIntentConfirmationHandler(
+            intentConfirmationInterceptor = intentConfirmationInterceptor,
+        ).apply {
+            registerWithCallbacks(
+                googlePayCallbackHandler = googlePayCallbackHandler,
+            )
+        }
+
+        intentConfirmationHandler.state.test {
+            assertThat(awaitItem()).isEqualTo(IntentConfirmationHandler.State.Idle)
+
+            intentConfirmationHandler.start(
+                arguments = DEFAULT_ARGUMENTS.copy(
+                    confirmationOption = GOOGLE_PAY_OPTION
+                ),
+            )
+
+            assertThat(awaitItem()).isEqualTo(IntentConfirmationHandler.State.Confirming(contentVisible = true))
+            assertThat(awaitItem()).isEqualTo(IntentConfirmationHandler.State.Confirming(contentVisible = false))
+
+            googlePayCallbackHandler.onResult(
+                GooglePayPaymentMethodLauncher.Result.Completed(
+                    paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD,
+                )
+            )
+
+            assertThat(awaitItem()).isEqualTo(IntentConfirmationHandler.State.Confirming(contentVisible = true))
+
+            val call = intentConfirmationInterceptor.calls.awaitItem()
+
+            assertThat(call).isEqualTo(
+                FakeIntentConfirmationInterceptor.InterceptCall.WithExistingPaymentMethod(
+                    initializationMode = PaymentSheet.InitializationMode.PaymentIntent(
+                        clientSecret = "pi_456_secret_456"
+                    ),
+                    paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD,
+                    paymentMethodOptionsParams = null,
+                    shippingValues = null,
+                )
+            )
+        }
+    }
+
     private fun createIntentConfirmationHandler(
         intentConfirmationInterceptor: IntentConfirmationInterceptor = FakeIntentConfirmationInterceptor(),
         bacsMandateConfirmationLauncher: BacsMandateConfirmationLauncher = FakeBacsMandateConfirmationLauncher(),
+        googlePayPaymentMethodLauncherFactory: GooglePayPaymentMethodLauncherFactory =
+            object : GooglePayPaymentMethodLauncherFactory {
+                override fun create(
+                    lifecycleScope: CoroutineScope,
+                    config: GooglePayPaymentMethodLauncher.Config,
+                    readyCallback: GooglePayPaymentMethodLauncher.ReadyCallback,
+                    activityResultLauncher: ActivityResultLauncher<GooglePayPaymentMethodLauncherContractV2.Args>,
+                    skipReadyCheck: Boolean
+                ): GooglePayPaymentMethodLauncher = mock()
+            },
```

Comment:
Ideally I would make an internal test class for `GooglePayPaymentMethodLauncher` however the class is public facing and has no implementable interface. I can follow-up with a test factory but I don't want to make this PR bigger than it already is.

Reply:
yeah this would be a great follow up! I think it'd be much nicer to be able to use a fake instead of a mock for the tests you added that rely on `GooglePayPaymentMethodLauncher`